src/main/java/dinkplugin/notifiers/PetNotifier.java

```diff
@@ -22,18 +37,52 @@ protected String getWebhookUrl() {
     }
 
     public void onChatMessage(String chatMessage) {
-        if (isEnabled() && PET_REGEX.matcher(chatMessage).matches()) {
-            this.handleNotify();
+        if (petName == null && isEnabled() && PET_REGEX.matcher(chatMessage).matches()) {
+            this.petName = "";
         }
     }
 
+    public void onClanNotification(String message) {
+        if (petName != null) {
+            Matcher matcher = CLAN_REGEX.matcher(message);
+            if (matcher.find()) {
+                String user = matcher.group("user").trim();
+                if (user.equals(Utils.getPlayerName(client))) {
+                    this.petName = matcher.group("pet");
+                }
+            }
+        }
+    }
+
+    public void onTick() {
+        if (petName != null)
+            this.handleNotify();
+    }
+
+    public void reset() {
+        this.petName = null;
+    }
+
     private void handleNotify() {
         String notifyMessage = config.petNotifyMessage()
             .replace("%USERNAME%", Utils.getPlayerName(client));
 
+        String thumbnail = Optional.ofNullable(petName)
+            .filter(s -> !s.isEmpty())
+            .map(Utils::ucFirst)
+            .map(itemSearcher::findItemId)
+            .map(ItemUtils::getItemImageUrl)
+            .orElse(null);
+
+        PetNotificationData extra = StringUtils.isNotEmpty(petName) ? new PetNotificationData(petName) : null;
```

Comment:
I think if we start sending `extra` out, we should always send it out and instead have the fields inside be optional

Got pet and figured out its name:
```json5
{
  "extra": {
    "petName": "forsen"
  }
```

Got pet but didn't figure out its name:

Alternative 1) null
```json5
{
  "extra": {
    "petName": null,
  }
```

Alternative 2) don't send the inner field
```json5
{
  "extra": {
  }
```

Alternative 3) send an empty string
```json5
{
  "extra": {
    "petName": "",
  }
```


@Leppunen this might be relevant since you've worked on your custom API, do you have a preference for any of the above examples (or am I maybe wrong in my entire assumption that `extra` should always be sent?)

Reply:
Went with Alternative 2